src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -2494,6 +2494,747 @@ End Class
+    End Sub
+
+    Shared Sub M6 (x as (Byte?, Byte?))
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M6 (x as (Short?, Short?))
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M7 (x as (Byte, Byte)?)
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M7 (x as (Short, Short)?)
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M8 (x as (Byte, Byte)?, y as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M8 (x as (Byte, Byte)?, y as Short)
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub Main()
+        M1(70000)
+        M2({ 300 }, 70000)
+        M3((70000, 70000))
+        M4((300, 300), 70000)
+
+        M5(70000)
+        M6((70000, 70000))
+        M7((70000, 70000))
+        M8((300, 300), 70000)
+    End Sub
+End Class
+    </file>
+</compilation>,
+options:=TestOptions.ReleaseExe.WithOverflowChecks(False), additionalRefs:=s_valueTupleRefs)
+
+            AssertTheseDiagnostics(comp, <expected></expected>)
+
+            CompileAndVerify(comp, expectedOutput:=
+"Byte
+Byte
+Byte
+Byte
+Byte
+Byte
+Byte
+Byte")
+        End Sub
+
+        <Fact>
+        Public Sub NarrowingFromNumericConstant_03()
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Option Strict On
+
+Public Class C
+
+    Shared Sub M1 (x as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M1 (x as Short)
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M2 (x as Byte(), y as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M2 (x as Byte(), y as Short)
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M3 (x as (Byte, Byte))
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M3 (x as (Short, Short))
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M4 (x as (Byte, Byte), y as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M4 (x as (Byte, Byte), y as Short)
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M5 (x as Byte?)
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M5 (x as Short?)
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M6 (x as (Byte?, Byte?))
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M6 (x as (Short?, Short?))
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M7 (x as (Byte, Byte)?)
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M7 (x as (Short, Short)?)
+        System.Console.WriteLine("Short")
+    End Sub
+
+    Shared Sub M8 (x as (Byte, Byte)?, y as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+    Shared Sub M8 (x as (Byte, Byte)?, y as Short)
+        System.Console.WriteLine("Short")
+    End Sub
+
```

Comment:
Perhaps add a couple nested cases like ```x as (Byte, (Byte, Byte))```   just to see the bit floats through more than one tuple